qiskit/ignis/verification/randomized_benchmarking/experiments/analysis.py

```diff
@@ -0,0 +1,665 @@
+
+        alpha = params[1]  # exponent
+        alpha_err = params_err[1]
+
+        nrb = 2 ** self.num_qubits()
+        epc = (nrb - 1) / nrb * (1 - alpha)
+        epc_err = (nrb - 1) / nrb * alpha_err / alpha
+
+        return RBResult({
+            'A': params[0],
+            'alpha': params[1],
+            'B': params[2],
+            'A_err': params_err[0],
+            'alpha_err': params_err[1],
+            'B_err': params_err[2],
+            'epc': epc,
+            'epc_err': epc_err,
+            'qubits': self._qubits,
+            'xdata': xdata,
+            'ydata': ydata,
+            'lengths': self._lengths,
+            'fit_function': self._rb_fit_fun,
+            'group_type': self._group_type
+        })
+
+class CNOTDihedralRBAnalysis(RBAnalysisBase):
+    def __init__(self,
+                 qubits: List[int],
+                 lengths: List[int],
+                 data: Optional[Union[BaseJob, Result, List[any], any]] = None,
+                 metadata: Optional[Union[List[Dict[str, any]], Dict[str, any]]] = None,
+                 name: Optional[str] = None,
+                 exp_id: Optional[str] = None
+                 ):
+        self._qubits = qubits
+        self._lengths = lengths
+        self.z_fitter = RBAnalysis(self._qubits, self._lengths)
+        self.x_fitter = RBAnalysis(self._qubits, self._lengths)
+        super().__init__(qubits, lengths, self.fit, data, metadata, name, exp_id, group_type='cnot_dihedral')
+
+    def add_data(self,
+                 data: Union[BaseJob, Result, any],
+                 metadata: Optional[Dict[str, any]] = None):
+        data, metadata = self.extract_data_and_metadata(data, metadata)
+        if data is not None:
+            z_metadata = [m for m in metadata if m['cnot_basis'] == 'Z']
+            x_metadata = [m for m in metadata if m['cnot_basis'] == 'X']
+            self.z_fitter.add_data(data, z_metadata)
+            self.x_fitter.add_data(data, x_metadata)
+
+    def run(self, **params) -> any:
+        z_fit_results = self.z_fitter.run()
+        x_fit_results = self.x_fitter.run()
+        self._result = self._analysis_fn(z_fit_results, x_fit_results)
+        return self._result
+
+    def fit(self, z_fit_results, x_fit_results):
+        # calculate nrb=d=2^n:
+        nrb = 2 ** len(self._qubits)
+
+        # Calculate alpha_Z and alpha_R:
+        alpha_Z = z_fit_results['alpha']
+        alpha_R = x_fit_results['alpha']
+        # Calculate their errors:
+        alpha_Z_err = z_fit_results['alpha_err']
+        alpha_R_err = x_fit_results['alpha_err']
+
+        # Calculate alpha:
+        alpha = (alpha_Z + nrb * alpha_R) / (nrb + 1)
+
+        # Calculate alpha_err:
+        alpha_Z_err_sq = (alpha_Z_err / alpha_Z / (nrb + 1)) ** 2
+        alpha_R_err_sq = (nrb * alpha_R_err / alpha_R / (nrb + 1)) ** 2
+        alpha_err = np.sqrt(alpha_Z_err_sq + alpha_R_err_sq)
+
+        # Calculate epg_est:
+        epg_est = (nrb - 1) * (1 - alpha) / nrb
```

Comment:
maybe one should change the name? it's not `epc` since it's not clifford, but also not `epg` since it's not a gate. 
maybe `epe` for an element?